rad/prelude/test.rad

```diff
@@ -1,73 +1,48 @@
-;; testing functions
+;; This file provides the `run-all-tests` function that runs the tests
+;; registered with the `:test` macro.
 
-(def is-test-env
-  "True iff file is being run as part of the Haskell suite"
-  (catch 'any (eq? test-env__ #t) (fn [x] #f)))
+(def run-test
+  "Run `test` and print the result. `test` is is an list of quoted
+  test cases with an optional `:setup` item.
+  ```
+  '(
+    [:setup (do
+      (def foo #t)
+    )]
+    [ foo ==> #t ]
+    [ (not foo) ==> #f ]
+  )
+  ```
+  "
+  (fn [test]
+    (def test-name (nth 0 test))
+    (def error
+      (fn [msg]
+        (put-str! (string-append "Test '" test-name "' failed: " msg))))
+    ;; Check a single test in a given env
+    (def check-single
```

Comment:
Maybe this is for a different PR, but it'd be nice if `run-test` caught exceptions - currently we get really weird failures if any test throws, and it's hard to figure out exactly what threw them. (Also, no *later* tests are run.)

Reply:
https://github.com/oscoin/radicle/issues/274